catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -384,4 +573,133 @@ private SchemaEntity updateSchemaEntity(
                 .build())
         .build();
   }
+
+  private String renameStorageLocation(
+      NameIdentifier nameIdent, FilesetEntity filesetEntity, String newName) {
+    if (StringUtils.isNotBlank(filesetEntity.storageLocation())) {
+      // For specified storageLocation, we should not rename the physical location.
+      return filesetEntity.storageLocation();
+
+    } else if (filesetEntity.filesetType().equals(Fileset.Type.MANAGED)) {
+      // For managed fileset with no storage location, we should also rename the physical location.
```

Comment:
Just another question: Should we support: 
1.  TTL for the external fileset
2. change an external fileset to a managed fileset

Reply:
> is it too expensive to rename the physical location? It depends on the performance of the storage. It is a bit slow when using an object storage like s3.
Maybe we can validate that the directory must be empty when creating a managed fileset. If the directory is not empty, we can only create an external fileset, because there may be a managed fileset on it.

Yeah, that's really an issue for object store. I tried to leverage the definition of Hive's manage table, Hive will rename the location to a new one when table is renamed. How about we don't do a directory renaming operation when we renaming the fileset? @coolderli 

> Another consideration, we may change the storage location for the managed fileset. For example, change the location from s3 to NAS so we can get better performance.

Yeah, we can support changing the storage location by adding more `FilesetChange` interfaces. Maybe we can move the folders when we explicitly change the storage location. Not implicitly when doing fileset rename, WDYT? @coolderli .

Reply:
> > is it too expensive to rename the physical location? It depends on the performance of the storage. It is a bit slow when using an object storage like s3.
> > Maybe we can validate that the directory must be empty when creating a managed fileset. If the directory is not empty, we can only create an external fileset, because there may be a managed fileset on it.
> 
> Yeah, that's really an issue for object store. I tried to leverage the definition of Hive's manage table, Hive will rename the location to a new one when table is renamed. How about we don't do a directory renaming operation when we renaming the fileset? @coolderli
> 

In my opinion, I'd like to not do a directory renaming operation when we rename the fileset. 
In Iceberg, renaming a table will just modify the metadata, so Iceberg is very suitable for object storage. I think the name of fileset is not necessarily the same as the physical path. Whether the fileset is managed or external,  the location specified by users may be different from the physical path. So we don't have to keep the consistency. This will make things simple.

Reply:
> > > is it too expensive to rename the physical location? It depends on the performance of the storage. It is a bit slow when using an object storage like s3.
> > > Maybe we can validate that the directory must be empty when creating a managed fileset. If the directory is not empty, we can only create an external fileset, because there may be a managed fileset on it.
> > 
> > 
> > Yeah, that's really an issue for object store. I tried to leverage the definition of Hive's manage table, Hive will rename the location to a new one when table is renamed. How about we don't do a directory renaming operation when we renaming the fileset? @coolderli
> 
> In my opinion, I'd like to not do a directory renaming operation when we rename the fileset. In Iceberg, renaming a table will just modify the metadata, so Iceberg is very suitable for object storage. I think the name of fileset is not necessarily the same as the physical path. Whether the fileset is managed or external, the location specified by users may be different from the physical path. So we don't have to keep the consistency. This will make things simple.

I see, let me think a bit on how to handle it.

Reply:
> > Another consideration, we may change the storage location for the managed fileset. For example, change the location from s3 to NAS so we can get better performance.
> 
> Yeah, we can support changing the storage location by adding more `FilesetChange` interfaces. Maybe we can move the folders when we explicitly change the storage location. Not implicitly when doing fileset rename, WDYT? @coolderli .


I'm not sure if we should move the folders when we explicitly change the storage location. If the storage is different storage such as s3 and hdfs, it may cost a lot of resources and time to finish it, I think it is beyond the scope of gravitino's management.

Reply:
I see. If we don't move the storage location when we update to a new location. It will be possible that some old data were in the old location and cannot be managed anymore. The thing why Iceberg doesn't move the location is that it has snapshot concept, and each version of metadata including the location is completely stored in the snapshot, so the old data can still be visited when location is updated.

Reply:
> I see. If we don't move the storage location when we update to a new location. It will be possible that some old data were in the old location and cannot be managed anymore. The thing why Iceberg doesn't move the location is that it has snapshot concept, and each version of metadata including the location is completely stored in the snapshot, so the old data can still be visited when location is updated.

That's really a problem. At first, we talked about if we should have snapshots in the fileset, but I think is too complex to record all files in metadata like Iceberg. This is also the question I want to ask, how to handle the migration from HDFS to s3 in the fileset.

In the past, I experienced a migration process from HDFS to OSS, which required a lot of time and resources. It is quite troublesome across different data centers. I also think we should not handle data copying across the data center unless we determine that it is in the same center. Maybe we should not ensure data consistency when the user updates the storage location. It's just like we update the hive table location.

Reply:
Currently, I think we can handle the migration manually, for example if we want to change the location fileset from A to B, we can manually move/copy the folder from A to B, then updating the fileset's location to point to B.

PS. updating fileset location can be done separately in another PR.